Sources/Sentry/SentrySession.m

```diff
@@ -186,6 +186,28 @@ - (void)incrementErrors
     }
 }
 
+- (id)copyWithZone:(nullable NSZone *)zone
+{
+    SentrySession *copy = [[[self class] allocWithZone:zone] init];
+
+    if (copy != nil) {
+        copy->_sessionId = _sessionId;
+        copy->_started = _started;
+        copy->_status = _status;
+        copy->_errors = _errors;
+        copy->_sequence = _sequence;
+        copy->_distinctId = _distinctId;
+        copy->_timestamp = _timestamp;
+        copy->_duration = _duration;
+        copy->_releaseName = _releaseName;
```

Comment:
Why do we use send_msg below (`.`) and deref on field on this one and above?
Shouldn't we use `[]` or `.` in all of them?

Reply:
We can't use `.` for most of them because they are read-only. We use `.` for `environment` and `user` because then we use the default setter which copies the values. FYI, I used AppCode to generate this code, but it is good that we question it still.